### ISSUE TITLE
[nodejs] `register_buffer()` registers only first table

### ISSUE THREAD

**Issue by pmm-motif** [duckdb/duckdb-node#32]
### What happens?

[register_buffer()](https://duckdb.org/docs/api/nodejs/reference#module_duckdb..Connection+register_buffer) function takes an `ArrowIterable` as argument for buffers to register. This implies that it's possible to register multiple of them under one table. However, when more than one is provided, only the first one is registered

### To Reproduce

Example - following code outputs "3" rather than "6" as expected

```typescript
import { tableFromArrays, tableToIPC } from 'apache-arrow';
import { Database } from 'duckdb';

const db = new Database(':memory:');
const sampleDataPart1 = tableFromArrays({
  a: [1, 2, 3],
});
const sampleDataPart2 = tableFromArrays({
  a: [4, 5, 6],
});

db.exec(` INSTALL arrow; LOAD arrow;`, (err) => {
  if (err) {
    throw err;
  }
  db.register_buffer(
    'sample_data',
    [tableToIPC(sampleDataPart1), tableToIPC(sampleDataPart2)],
    true,
    (err) => {
      if (err) {
        throw err;
      }
      db.all('select count(*) as total from sample_data', (err, res) => {
        if (err) {
          throw err;
        }
        console.log((res[0] as any).total);
      });
    }
  );
});
```

### OS:

macos

### DuckDB Version:

0.8.1

### DuckDB Client:

NodeJS

### Full Name:

Przemek Maciolek

### Affiliation:

Motif Analytics

### Have you tried this on the latest `main` branch?

I have tested with a main build

### Have you tried the steps to reproduce? Do they include all relevant data and configuration? Does the issue you report still appear there?

- [X] Yes, I have

**Comment by pmm-motif** [comment]
I finally had the opportunity to spend some time on this issue myself. I originally thought that maybe it's something trivial to fix (like only the first array element is passed downstream) but realized that it's not the case. Finally, got into [duckdb/arrow](https://github.com/duckdb/arrow). Overall, found some interesting bits there, but maybe not directly related to this project (bringing these separately below)

TLDR, it looks like `ArrowIterable` must represent a single Arrow table. While all buffers are passed, after reading EOS no more batches seem to be added by the [stream decoder](https://github.com/duckdb/arrow/blob/main/src/arrow_scan_ipc.cpp#L35). I think it's by design and changing it would be somewhat heavy. Perhaps two things should be changed then:
* mention in the docs/API that `ArrowIterable` always represents a single Arrow Table
* throw an exception when more batches after reading EOS

**Side questions**

1) While most of the projects refer to DuckDB core via git submodule, `duckdb-node` is using [a tree copy...](https://github.com/duckdb/duckdb-node/tree/main/src/duckdb). Just wondering, is that a temporary thing?

2) It looks that [arrow_scan_ipc.cpp](https://github.com/duckdb/arrow/blob/main/src/arrow_scan_ipc.cpp) is largely duplicating [arrow.cpp](https://github.com/duckdb/duckdb/blob/main/src/function/table/arrow.cpp) in DuckDB core. I guess it might be an interim state, though if that's the case I wonder when this code is supposed to migrate?

**Comment by acuitymd-filip** [comment]
Kind of a bummer for Node specifically, as this effectively means that there is a hard limit on the size of Arrow data that can be registered with DuckDB due to Node's [limit on the max length of TypedArrays](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Errors/Invalid_array_length). So on a 64bit system the max size of Arrow data that Node DuckDB can work with is about 4GB. Anyone has found any workarounds for this?
